extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -214,7 +215,51 @@ export class VariantAnalysisManager
   }
 
   private async runVariantAnalysisFromPublishedPack(): Promise<void> {
-    throw new Error("Command not yet implemented");
+    const language = await askForLanguage(this.cliServer);
+
+    const packName = `codeql/${language}-queries`;
+    const packDownloadResult = await this.cliServer.packDownload([packName]);
+    const downloadedPack = packDownloadResult.packs[0];
+
+    const packDir = `${packDownloadResult.packDir}/${downloadedPack.name}/${downloadedPack.version}`;
+
+    const suitePath = `${packDir}/codeql-suites/${language}-code-scanning.qls`;
+    const resolvedQueries = await this.cliServer.resolveQueries(suitePath);
+
+    const problemQueries =
+      await this.filterToOnlyProblemQueries(resolvedQueries);
+
+    if (problemQueries.length === 0) {
+      void this.app.logger.showErrorMessage(
+        `Unable to trigger variant analysis. No problem queries found in published query pack: ${packName}.`,
+      );
+    }
+
+    return withProgress((progress, token) =>
```

Comment:
I haven't tried running this yet, but would it be helpful to start the progress earlier since quite some work is being done is this method already (downloading the pack, resolving the queries and then resolving query metadata for all queries)?